tests/devices/unit_tests/test_shutter.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+from ophyd_async.core import DeviceCollector
+
+from dodal.devices.shutter import OpenState, Shutter
+
+
+@pytest.fixture
+async def sim_shutter():
+    async with DeviceCollector(sim=True):
+        sim_shutter: Shutter = Shutter(
+            prefix="sim_shutter",
+            name="shutter",
+        )
+    return sim_shutter
+
+
+@pytest.mark.parametrize(
+    "state",
+    [
+        (OpenState.OPEN),
+        (OpenState.CLOSE),
+    ],
+)
+async def test_set_opens_and_closes_shutter(state: OpenState, sim_shutter: Shutter):
+    status = sim_shutter.set(state)
+    assert not status.done
```

Comment:
Why assert not status.done?

Reply:
as that is what was in the `sample_shutter` test:
`import pytest
from ophyd.sim import make_fake_device

from dodal.devices.sample_shutter import OpenState, SampleShutter


@pytest.fixture
def fake_sample_shutter():
    FakeSampleShutter = make_fake_device(SampleShutter)
    fake_sample_shutter: SampleShutter = FakeSampleShutter(name="sample_shutter")
    return fake_sample_shutter


def set_pos_rbv(fake_sample_shutter: SampleShutter, value):
    fake_sample_shutter.pos_rbv.sim_put(value)  # type: ignore


@pytest.mark.parametrize(
    "state",
    [
        (OpenState.OPEN),
        (OpenState.CLOSE),
    ],
)
def test_set_opens_and_closes_shutter(state, fake_sample_shutter):
    status = fake_sample_shutter.set(state)
    assert not status.done
    set_pos_rbv(fake_sample_shutter, state.value)
    status.wait(1)
`

I don't know much about shutters, so kept that behavior

Reply:
`status` should be awaited at the end of the test then, to avoid leakage, which also preserves the original behavoir

Reply:
that was just an example, not this PR

Reply:
It still needs to be awaited

Reply:
ok, done

Reply:
Unresolving as I think this will be non-deteministic in the tests. 

I think we should await the status here instead: it's the status for the move, if we're not mocking out the move, then we can't reliably say that it hasn't completed already; we're also immediately after reading the signal and asserting that its value is the value that we have told it to move to (i.e. that the move has completed). We should instead request the move, await the move, then assert that we have moved to the requested position.

Reply:
not sure how to apply which after the change